pkg/operator/apis/monitoring/v1alpha1/types.go

```diff
@@ -510,28 +514,52 @@ func (pm *PodMonitoring) endpointScrapeConfig(index int) (*promconfig.ScrapeConf
 			TargetLabel:  "instance",
 		})
 	} else if ep.Port.IntVal != 0 {
+		// Prometheus produces a target candidate for each declared port in a pod. If a pod
+		// has no declared ports, a single candidate without port info is generated.
+		// Some users do not declare ports but expect PodMonitoring's to work with a numeric port
+		// specified nonetheless.
+		//
+		// We can support this by hard-overriding the address with the respective numeric port. If a pod
+		// has multiple declared ports that will produce multiple identical targets from the candidates. If
+		// the container metadata label is added however, they may differ by the `container` target label
+		// and not be deduplicated. This will cause the same endpoint to be scraped as different targets.
+		// Thus, we must still filter by numeric port IFF the target candidate does contain port info.
+		//
+		// Summarized, PodMonitorings with numeric ports always work if a pod has no declared ports.
```

Comment:
Should we echo this in the API docs by putting it [here](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/0df48f25673168ee01ca12c8d41114e93f58b43d/pkg/operator/apis/monitoring/v1alpha1/types.go#L1072) in the context of a `ScrapeEndpoint`?